src/main/groovy/wooga/gradle/unity/UnityPlugin.groovy

```diff
@@ -67,46 +67,28 @@ class UnityPlugin implements Plugin<Project> {
         this.project = project
         project.pluginManager.apply(BasePlugin.class)
         project.pluginManager.apply(ReportingBasePlugin.class)
-        UnityPluginExtension extension = project.extensions.create(EXTENSION_NAME, DefaultUnityPluginExtension, project, fileResolver, instantiator)
 
+        final UnityPluginExtension unityExtension = project.extensions.create(EXTENSION_NAME, DefaultUnityPluginExtension, project, fileResolver, instantiator)
         final ReportingExtension reportingExtension = (ReportingExtension) project.getExtensions().getByName(ReportingExtension.NAME)
-        ConventionMapping unityExtensionMapping = ((IConventionAware) extension).getConventionMapping()
-        unityExtensionMapping.map("reportsDir", new Callable<Object>() {
-            @Override
-            Object call() {
-                return reportingExtension.file("unity")
-            }
-        })
-
-        unityExtensionMapping.map("assetsDir", new Callable<Object>() {
-            @Override
-            Object call() {
-                return new File(extension.getProjectPath().path, "Assets")
-            }
-        })
-
-        unityExtensionMapping.map("pluginsDir", new Callable<Object>() {
-            @Override
-            Object call() {
-                return new File(extension.getAssetsDir(), "Plugins")
-            }
-        })
-
-        BasePluginConvention convention = new BasePluginConvention(project)
+        final ConventionMapping unityExtensionConvention = ((IConventionAware) unityExtension).getConventionMapping()
+        final BasePluginConvention basePluginConvention = new BasePluginConvention(project)
 
+        unityExtensionConvention.map("reportsDir", { return reportingExtension.file("unity") })
+        unityExtensionConvention.map("assetsDir", { return new File(unityExtension.getProjectPath().path, "Assets") })
+        unityExtensionConvention.map("pluginsDir", { return new File(unityExtension.getAssetsDir(), "Plugins") })
```

Comment:
You don't need the return statements in the closures.
My old code was more Java style. I think we should find a good balance. 

And we should extract the logic into a method.